File: front-end/react-app/src/utils/post.ts


// TODO: The backend is returning the whole post class, which is not necessary for the front-end.
export interface Post {
    postId: string;
    userId: number;
    title: string;
    content: string;
    isArchived: boolean;
    status: 'PUBLISHED' | 'UNPUBLISHED';
    dateCreated: string;
    dateModified: string;
    images: string[]; 
    attachments: string[];
    postReplies: null | PostReply[]; // Assuming it could be an array of replies or null, specify further if possible
  }

export interface PostReply {
    replyId: string | null; // Assuming replyId can be null based on the given structure
    userId: number | null; // Assuming userId can be null
    comment: string;
    isActive: boolean;
    dateCreated: string;
    subReplies: SubReply[] | null; // Array of SubReply objects or null
}
  
export interface SubReply {
    subReplyId: string | null; // Assuming subReplyId can be null
    userId: number | null; // Assuming userId can be null
    comment: string;
    isActive: boolean;
    dateCreated: string;
}

File: back-end/Microservice/CompositeService/src/main/java/com/example/compositeservice/entity/remote/Post.java
package com.example.compositeservice.entity.remote;

import lombok.*;

import java.util.Date;
import java.util.List;


@Getter
@Setter

public class Post {
    private String postId;
    private Long userId;
    private String title;
    private String content;
    private Boolean isArchived;
    private StatusEnum status;
    private Date dateCreated;
    private Date dateModified;
    private List<String> images;
    private List<String> attachments;
    private List<PostReply> postReplies;
}
